src/Stratis.Bitcoin.Features.PoA/Voting/IdleFederationMembersKicker.cs

```diff
@@ -119,14 +119,18 @@ public bool ShouldBeKicked(PubKey pubKey, uint blockTime, out uint inactiveForSe
                 inactiveForSeconds = 0;
                 return false;
             }
-            
+
             inactiveForSeconds = blockTime - lastActiveTime;
 
             return (inactiveForSeconds > this.federationMemberMaxIdleTimeSeconds && !this.federationManager.IsMultisigMember(pubKey));
         }
 
         private void OnBlockConnected(BlockConnected blockConnectedData)
         {
+            // Do nothing if this is not a mining node on Cirrus.
+            if (this.federationManager is FederationManager)
```

Comment:
Perhaps a negative test would be better: e.g. `if (!(this.federationManager is CollateralFederationManager))` ?